src/Models/ShallowWaterModels/solution_and_tracer_tendencies.jl

```diff
@@ -2,19 +2,10 @@ using Oceananigans.Advection
 using Oceananigans.Coriolis
 using Oceananigans.Operators
 
-@inline squared(i, j, k, grid, ϕ) = ϕ[i, j, k]^2
+@inline squared(i, j, k, grid, ϕ) = @inbounds ϕ[i, j, k]^2
 
-@inline uh_transport_x(i, j, k, grid, h, uh, g) =
-    @inbounds ℑxᶜᵃᵃ(i, j, k, grid, squared, uh) / h[i, j, k] + g/2 * h[i, j, k]^2
-
-@inline uh_transport_y(i, j, k, grid, h, uh, vh) =
-    ℑyᵃᶠᵃ(i, j, k, grid, uh) * ℑxᶠᵃᵃ(i, j, k, grid, vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
-
-@inline vh_transport_x(i, j, k, grid, h, uh, vh) =
-    ℑyᵃᶠᵃ(i, j, k, grid, uh) * ℑxᶠᵃᵃ(i, j, k, grid, vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
-
-@inline vh_transport_y(i, j, k, grid, h, vh, g) =
-    @inbounds ℑyᵃᶜᵃ(i, j, k, grid, squared, vh) / h[i, j, k] + g/2 * h[i, j, k]^2
+@inline gh2(i, j, k, grid, h, g) =
+    @inbounds g/2 * h[i, j, k]^2
```

Comment:
This function could probably be defined on a single line.

Reply:
Done